### PR TITLE
Cleanup docstring types

### LINKED ISSUE #10: Custom parser for type hints
Type hints are nice to check the code but it makes the documentation hard to read.
We should write a custom parser to display the hints in a cleaner way.

Related: https://michaelgoerz.net/notes/extending-sphinx-napoleon-docstring-sections.html

and https://github.com/agronholm/sphinx-autodoc-typehints

Update after #167: Above added. Remaining TODOs:
- [ ] Clean up parentheses in the type hinting (they are formatted as standard text, not in code-style).

Regex used:
```regex
(:((param)|(return))\s*[a-zA_z_0-9]*:) (\([a-zA-Z_\[\]\s,\.0-9]+\))
```
and replace with `$1`.

### PR DESCRIPTION
**Pull request by araffin**
## Description

closes #10  (follow up of #167 )

I used some regex magic to do that ;) (see details in the issue).
I used some css to make the type prettier.
The result: https://stable-baselines3.readthedocs.io/en/doc-clean-types/modules/base.html

TODO:
- [x] remove duplicate types
- [x] Post-process the output of the sphinx autodoc extension (see discussion in #167 )

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by araffin** [comment]
Please wait a tiny bit before merging... I'm experimenting a better way to solve the style issue

**Comment by araffin** [comment]
It is ready now ;) (I tested a js hack, that was kinda working but unsatisfying)
I also trying playing with sphinx extension but could not manage to change the formatting without breaking the build.

File: stable_baselines3/common/save_util.py [hunk 1]
```diff
@@ -360,11 +360,11 @@ def load_from_zip_file(
...
     Load model data from a .zip archive
 
-    :param load_path: (str, pathlib.Path, io.BufferedIOBase) Where to load the model from
+    :param load_path: Where to load the model from
     :param load_data: Whether we should load and return data
         (class parameters). Mainly used by 'load_parameters' to only load model parameters (weights)
-    :param device: (Union[th.device, str]) Device on which the code should run.
-    :return: (dict),(dict),(dict) Class parameters, model state_dicts (aka "params", dict of state_dict)
+    :param device: Device on which the code should run.
+    :return:,(dict),(dict) Class parameters, model state_dicts (aka "params", dict of state_dict)
```
Comment (Miffyli) [inline stable_baselines3/common/save_util.py:367]: This seems bit wonky

**Comment by Miffyli** [comment]
Skimmed through results and looks good to me, apart from one derp-looking docstring :)

**Comment by Miffyli** [comment]
Is this ready for checking/merging? Things look good but I wonder if there are still some docstrings missing :)

**Comment by araffin** [comment]
> Is this ready for checking/merging? Things look good but I wonder if there are still some docstrings missing :)

ready for what it was meant for ;)

**Comment by araffin** [comment]
finishing to type everything is another issue

**Review by Miffyli — APPROVED** [review]
In that case LGTM! Will take an internal note on cleaning up any remaining types in doctstrings / adding missing ones :)

**Pull request merged**